crates/fluvio-smartengine/Cargo.toml

```diff
@@ -33,7 +33,7 @@ dataplane = { version = "0.11.0", path = "../fluvio-dataplane-protocol", package
 ] }
 fluvio-controlplane-metadata = { path = "../fluvio-controlplane-metadata", version = "0.17.0", optional = true }
```

Comment:
I think this requires major bump with enum changes

Reply:
if we bump up the major version, maybe we can deprecate other manual smart module types

Reply:
I did the bump, not sure what is the expected approach to deprecate those manual smart modules

Reply:
k. let's keep as is